portability-transfer/src/main/java/org/dataportabilityproject/transfer/PortabilityInMemoryDataCopier.java

```diff
@@ -121,16 +124,57 @@ private void copyHelper(
             exportAuthData,
             importAuthData,
             Optional.of(new ExportInformation(
-                continuationData.getPaginationData(), exportInformation.get().getContainerResource())));
+                continuationData.getPaginationData(),
+                exportInformation.get().getContainerResource())));
       }
 
       // Start processing sub-resources
       if (continuationData.getContainerResources() != null
           && !continuationData.getContainerResources().isEmpty()) {
         for (ContainerResource resource : continuationData.getContainerResources()) {
-          copyHelper(jobId, exportAuthData, importAuthData, Optional.of(new ExportInformation(null, resource)));
+          copyHelper(jobId, exportAuthData, importAuthData,
+              Optional.of(new ExportInformation(null, resource)));
+        }
+      }
+    }
+  }
+
+  private ExportResult export(UUID jobId, AuthData exportAuthData,
+      Optional<ExportInformation> exportInformation) {
+    ExportResult<?> exportResult = exporter.get().export(jobId, exportAuthData, exportInformation);
+    for (int attempts = 0; exportResult.getType() == ResultType.ERROR; attempts++) {
+      ExceptionType response = checkRetry(exportResult.getMessage(), attempts);
```

Comment:
is ExceptionType response used for anything besides checking whether to retry? 

If not, I'm wondering if checkRetry can just return a boolean instead

Reply:
That's what I was thinking.  I believe that ultimately something like ExceptionType will also be holding retries, but I think that can wait until v1.